### ISSUE TITLE
getMarkdownHeaders "headers" regex null

### ISSUE THREAD

**Issue by jeppe1990** [mulesoft/api-console#248]
I have a RAML file:
![cusersjbdocumentsraml1stexperiment raml_-_su_2016-03-31_13-27-26](https://cloud.githubusercontent.com/assets/17174187/14174407/65050db0-f744-11e5-835c-386496eb3eac.png)

But the documentation section in api console doesn't show the content; just the titles:
![api_console_ _google_chrome_2016-03-31_13-29-54](https://cloud.githubusercontent.com/assets/17174187/14174465/b5604388-f744-11e5-92db-013c43c13f7b.png)

I narrowed down the problem to the getMarkdownHeaders function:
![api_console_ _google_chrome_2016-03-31_13-32-21](https://cloud.githubusercontent.com/assets/17174187/14174511/1e5d2838-f745-11e5-8fb2-905c82f5c784.png)

If I change the function so that
`headers = true`
it throws a few exceptions but it shows the content of the documentation:
![api_console_ _google_chrome_2016-03-31_13-35-33](https://cloud.githubusercontent.com/assets/17174187/14174581/902696a2-f745-11e5-8b84-9b3503e59cce.png)

Which isn't exactly pretty.

Are there some special characters I need to put in the beginning of my content?

PS: I have virtually zero experience with Regex.

**Comment by dmartinezg** [comment]
Hi @jeppe1990, thanks for looking into this, @juancoen, @xaka, can you guys give me a hand looking over this?

**Comment by juancoen** [comment]
Hi @jeppe1990, that function tries to match Markdown headers (lines starting with `#`) to display them as sub-sections below each title of the documentation. If no headers are found, to see the content of a documentation section, a user has to click on the title (in your example, clicking on `Home` will display the content of that documentation section).

I've flagged this as a `UX` improvement as it can be confusing.

Thanks!

**Comment by jeppe1990** [comment]
In the image I included showing the UI ("home" and "legal") I actually clicked on the Home tab before making the screenshot. There is simply no text, if markdown headers are not present. 
I believe that this must be a mistake since the RAML spec concerning the documentation section only use plain text and no specific markdown syntax in the "content" section (it's simply text): [RAML spec concerning User documentation](https://github.com/raml-org/raml-spec/blob/raml-10/versions/raml-10/raml-10.md/#user-documentation)
In other words, for text to show up in the documentation section of API Console it seems that text must contain symbols, that are not required by the RAML spec. 
That's not really a problem if this was evident in the API Console documentation.

**Comment by jarrodek** [comment]
Fixed with https://github.com/mulesoft/api-console/commit/c74eacde58e40d1d79c4885b15fa3ff210b6610c
